jenkins_jobs/modules/publishers.py

```diff
@@ -2456,37 +2456,119 @@ def base_email_ext(registry, xml_parent, data, ttype):
+                    ).text = ""
+                elif "recipients" == recipient:
+                    if "recipient_providers" not in locals():
+                        recipient_providers = XML.SubElement(
+                            email, "recipientProviders"
+                        )
+                    XML.SubElement(
+                        recipient_providers,
+                        "hudson.plugins.emailext.plugins.recipients.ListRecipientProvider",
+                    ).text = ""
+                elif "failing-test-suspects-recipients" == recipient:
+                    if "recipient_providers" not in locals():
+                        recipient_providers = XML.SubElement(
+                            email, "recipientProviders"
+                        )
+                    XML.SubElement(
+                        recipient_providers,
+                        "hudson.plugins.emailext.plugins.recipients.FailingTestSuspectsRecipientProvider",
+                    ).text = ""
+                elif "first-failing-build-suspects-recipients" == recipient:
+                    if "recipient_providers" not in locals():
+                        recipient_providers = XML.SubElement(
+                            email, "recipientProviders"
+                        )
+                    XML.SubElement(
+                        recipient_providers,
+                        "hudson.plugins.emailext.plugins.recipients.FirstFailingBuildSuspectsRecipientProvider",
+                    ).text = ""
+
+            # `failureCount` is deprecated and has no effect
+            # on email, but the element has been created
+            # in order to match the XML generated via UI.
+            failure_count_supporters = ["FirstFailureTrigger", "SecondFailureTrigger"]
+            if ttype in failure_count_supporters:
+                XML.SubElement(trigger, "failureCount").text = "0"
     else:
-        XML.SubElement(email, "sendToRequester").text = "false"
-        XML.SubElement(email, "sendToDevelopers").text = "false"
-        XML.SubElement(email, "includeCulprits").text = "false"
-        XML.SubElement(email, "sendToRecipientList").text = "true"
+        if plugin_version < pkg_resources.parse_version("2.39"):
+            XML.SubElement(email, "sendToRequester").text = "false"
+            XML.SubElement(email, "sendToDevelopers").text = "false"
+            XML.SubElement(email, "includeCulprits").text = "false"
+            XML.SubElement(email, "sendToRecipientList").text = "true"
     if ttype == "ScriptTrigger":
         XML.SubElement(trigger, "triggerScript").text = data["trigger-script"]
 
+    if plugin_version >= pkg_resources.parse_version("2.39"):
+        mappings = [
+            ("attachments", "attachmentsPattern", ""),
+            ("attach-build-log", "attachBuildLog", False),
+            ("compress-log", "compressBuildLog", False),
```

Comment:
I guess these new API keys should be documented.

Reply:
They are not changeable values anymore, therefore I don't think we need to document them.